future/src/ct/ct_filesystem.cc

```diff
@@ -0,0 +1,341 @@
+bool exists(const path& filepath) {
+    return Glib::file_test(filepath.string(), Glib::FILE_TEST_EXISTS);
+}
+
+// Open Filepath with External App
+void external_filepath_open(const fs::path& filepath, bool open_folder_if_file_not_exists, CtConfig* config)
+{
+    spdlog::debug("filepath to open: {}", filepath);
+    if (config->filelinkCustomOn) {
+        std::string cmd = fmt::sprintf(config->filelinkCustomAct, filepath.string());
+        std::system(cmd.c_str());
+    } else {
+        if (open_folder_if_file_not_exists && !fs::exists(filepath)) {
+            external_folderpath_open(filepath, config);
+        } else if (!fs::exists(filepath)) {
+            throw std::runtime_error(fmt::format("Filepath: {} does not exist and open_folder_if_not_exists is false", filepath.string()));
+        } else {
+#ifdef _WIN32
+            ShellExecute(GetActiveWindow(), "open", filepath.c_str(), NULL, NULL, SW_SHOWNORMAL);
+#else
+            fs::path f_path("file://");
+            f_path += filepath;
+            g_app_info_launch_default_for_uri(f_path.c_str(), nullptr, nullptr);
+#endif
+        }
+    }
+}
+
+// Open Folderpath with External App
+void external_folderpath_open(const fs::path& folderpath, CtConfig* config)
+{
+    spdlog::debug("dir to open: {}", folderpath.string());
+    if (config->folderlinkCustomOn) {
+        std::string cmd = fmt::sprintf(config->filelinkCustomAct, folderpath.string());
+        std::system(cmd.c_str());
+    } else {
+
+        // https://stackoverflow.com/questions/42442189/how-to-open-spawn-a-file-with-glib-gtkmm-in-windows
+#ifdef _WIN32
+        ShellExecute(NULL, "open", folderpath.c_str(), NULL, NULL, SW_SHOWDEFAULT);
+#elif defined(__APPLE__)
+        std::vector<std::string> argv = { "open", folderpath.string() };
+Glib::spawn_async("", argv, Glib::SpawnFlags::SPAWN_SEARCH_PATH);
+#else
+        fs::path path("file://");
+        path += folderpath;
+        g_app_info_launch_default_for_uri(folderpath.c_str(), nullptr, nullptr);
+#endif
+    }
+}
+
+path prepare_export_folder(const path& dir_place, path new_folder, bool overwrite_existing)
+{
+    if (fs::is_directory(dir_place / new_folder))
+    {
+        // todo:
+        if (overwrite_existing) {
+            spdlog::debug("removing dir: {}", dir_place / new_folder);
+            remove_all(dir_place / new_folder);
+        }
+        else {
+            int n = 2;
+            while (fs::is_directory(dir_place / (new_folder.string() + str::format("{:03d}", n))))
+                n += 1;
+            new_folder += str::format("{:03d}", n);
+        }
+    }
+    return new_folder;
+}
+
+std::uintmax_t remove_all(const path& dir)
+{
+    std::uintmax_t count = 0;
+    for (const auto& file : get_dir_entries(dir)) {
+        ++count;
+        if (is_directory(file)) {
+            count += remove_all(file);
+        }
+        remove(file);
+    }
+    remove(dir);
+    ++count;
+    return count;
+}
+
+std::string get_cherrytree_datadir()
```

Comment:
why not fs::path?

Reply:
I see `fs::path get_cherrytree_localedir()` with fs::path, but others are std::string

Reply:
Must have missed those, I'll correct